backends/bmv2/common/helpers.h

```diff
@@ -58,6 +58,17 @@ class V1ModelProperties {
     static const cstring validField;
 };
 
+// V1Model-specific blocks.
+enum class BlockConverted {
```

Comment:
Is the fact that BlockConverted is v1model-specific part of what you consider the "leakiness of the abstraction" mentioned in your comment on this PR?

I am not familiar with the overall structure of the code here, but agree that lifting out common stuff seems like a good idea.   I guess one question is how users of this code that are not implementing the v1model architecture replace this with something else.  Perhaps that is something you are hoping is addressed with later PRs?

Reply:
>Is the fact that BlockConverted is v1model-specific part of what you consider the "leakiness of the abstraction" mentioned in your comment on this PR?

Yeah, there are few other parts too. Like the meter map that needs to be part of the top level. Currently, the ProgramStructure still has elements that are v1model-specific. Rewriting code to make ProgramStructure fully generic is a little bit involved, so I am not addressing it in this PR.